sdk/ml/azure-ai-ml/azure/ai/ml/entities/_validation.py

```diff
@@ -175,27 +178,42 @@ def __init__(self, target_obj: Optional[typing.Dict[str, typing.Any]] = None):
 
     def merge_with(
         self,
-        target: "MutableValidationResult",
+        target: typing.Union[ValidationResult, DeploymentValidateResult],
         field_name: Optional[str] = None,
         condition_skip: Optional[typing.Callable] = None,
+        overwrite: bool = False,
     ):
         """Merge errors & warnings in another validation results into current one.
 
         Will update current validation result.
         If field_name is not None, then yaml_path in the other validation result will be updated accordingly.
         * => field_name, jobs.job_a => field_name.jobs.job_a e.g.. If None, then no update.
 
-        :param target: Validation result to merge.
-        :type target: MutableValidationResult
+        :param target: Validation result to merge. Also accept DeploymentValidateResult.
```

Comment:
When will `target` is instance of `DeploymentValidateResult`?

Reply:
`DeploymentValidationResult` is deserialized from the response of remote validation call. And per previous discussion, all resource validation will use the same contract. So here `deployment` means deployment of any type of resources